staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -608,6 +608,20 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 	if c.SharedInformerFactory != nil && !s.isPostStartHookRegistered(genericApiServerHookName) {
 		err := s.AddPostStartHook(genericApiServerHookName, func(context PostStartHookContext) error {
 			c.SharedInformerFactory.Start(context.StopCh)
+
+			// Wait for all caches to sync.
+			started := c.SharedInformerFactory.WaitForCacheSync(context.StopCh)
```

Comment:
For unit-testability, consider putting this functionality into a separate function?